src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -379,6 +380,10 @@ private Pair<Boolean, AtomicInteger> addNewLogEvent(AtomicInteger totalBytesRead
             return new Pair<>(true, new AtomicInteger());
         }
 
+        if (attemptLogInformation.getLogEvents().size() >= MAX_NUM_OF_LOG_EVENTS) {
```

Comment:
```suggestion
        if (attemptLogInformation.getLogEvents().size() > MAX_NUM_OF_LOG_EVENTS) {
```
From linked docs it seems like 10000 is allowed? 
> The maximum number of log events in a batch is 10,000

Reply:
If `attemptLogInformation.getLogEvents()` is already 10000, it should return that the max limit is reached. In fact, this check need not compare if it is greater than 10,000 as we only add one event at a time after this check. 

https://github.com/aws-greengrass/aws-greengrass-log-manager/blob/fd906cb0c1dc2f23d0a054a98965dee73b2ed710/src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java#L420

Reply:
Ah makes sense-- I scanned the code too quickly 😅